server/datastore/mysql/apple_mdm.go

```diff
@@ -2316,17 +2321,30 @@ func (ds *Datastore) bulkSetPendingMDMAppleHostProfilesDB(
 		}
 	}
 
+	// Loop through the current profiles and update the undesired ones that require removal.
 	for _, p := range currentProfiles {
 		if _, ok := profileIntersection.GetMatchingProfileInDesiredState(p); ok {
+			// We already handled upserting the current profiles in the desired state above.
 			continue
 		}
+
+		if p.OperationType == fleet.MDMOperationTypeRemove {
+			// At this point, we don't need further updates to any other removals that are already in the
+			// database. They should be allowed to run their course.
+			continue
+		}
+
 		// If the profile wasn't installed, then we do not want to change the operation to "Remove".
 		// Doing so will result in Fleet attempting to remove a profile that doesn't exist on the
 		// host (since the installation failed). Skipping it here will lead to it being removed from
 		// the host in Fleet during profile reconciliation, which is what we want.
 		if p.DidNotInstallOnHost() {
+			// TODO: maybe we want to delete the row here instead of waiting for the reconciliation?
```

Comment:
IIRC, we skipped deleting the row here in order to leave that logic in the service layer. I could see it being done here tho, I can't think of a reason why it would break anything right now.